underfs/cos/src/main/java/alluxio/underfs/cos/COSOutputStream.java

```diff
@@ -158,10 +158,15 @@ public void close() throws IOException {
         meta.setContentMD5(new String(Base64.encodeBase64(hashBytes)));
       }
       mCosClient.putObject(mBucketName, mKey, in, meta);
-      mFile.delete();
     } catch (CosClientException e) {
       LOG.error("Failed to upload {}. Temporary file @ {}", mKey, mFile.getPath());
       throw new IOException(e);
+    } finally {
+      // Delete the temporary file on the local machine if the transfer manager completed the
```

Comment:
Since we delete in the finally, we should change the error message above (the temp file will no longer be available)